docs/installation.rst

```diff
@@ -20,6 +20,35 @@ There are different ways to install |project_name|:
     have the latest code or use revision control (each release is
     tagged).
 
+.. _installation-requirements:
+
+Pre-Installation Considerations
+-------------------------------
+
+Repository Filesystem
+~~~~~~~~~~~~~~~~~~~~~
+:ref:data-structures-and-file-formats
+- File names may contain any valid ascii character and be of any length. |project_name|
+  stores data only 3 directory levels deep and uses short directory and file names.
+- |project_name| requires read and write permissions on the repository file system.
+- |project_name| stores backup metadata and data into so-called segment files. The 
+  target size of these files and also the count of these files per directory is set 
+  in the :ref:config-file.
+- |project_name| uses Mkdir based `file locking<https://en.wikipedia.org/wiki/File_locking#Lock_files`_.
```

Comment:
"""
... uses a generic and very portable mkdir-based locking mechanism.
It does **not use**: POSIX locks, NFS locks, windows file locks, lockf(), flock(), hardlinks.
"""
Pointing out what it does not use is useful here because these mechanisms often cause issues (and we have seen quite some of them in attic).